pkg/machine/hyperv/machine.go

```diff
@@ -170,12 +174,39 @@ func (m *HyperVMachine) Init(opts machine.InitOptions) (bool, error) {
 		Name:      user,
 		Key:       key,
 		VMName:    m.Name,
+		VMType:    machine.HyperVVirt,
 		TimeZone:  opts.TimeZone,
 		WritePath: m.IgnitionFile.GetPath(),
 		UID:       m.UID,
 	}
 
-	if err := machine.NewIgnitionFile(ign, machine.HyperVVirt); err != nil {
+	if err := ign.GenerateIgnitionConfig(); err != nil {
+		return false, err
+	}
+
+	// ready is a unit file that sets up the virtual serial device
+	// where when the VM is done configuring, it will send an ack
+	// so a listening host knows it can being interacting with it
+	//
+	// VSOCK-CONNECT:2 <- shortcut to connect to the hostvm
```

Comment:
if a user were wanting to run FCOS on hyperv outside of podman-machine, would they need something like this too?

Reply:
no, you can just use a regular network connection and you are good to go.

Reply:
ok, so in this case we're wanting to do something special just for podman machine? i.e. create a communication channel with the host?